runtime-transaction/src/compute_budget_instruction_details.rs

```diff
@@ -17,6 +18,24 @@ use {
     std::num::NonZeroU32,
 };
 
+#[cfg_attr(test, derive(Eq, PartialEq))]
+#[cfg_attr(feature = "dev-context-only-utils", derive(Clone))]
+#[derive(Debug)]
+struct MigrationBuiltinFeatureCounter {
+    // The vector of counters, matching the size of the static vector MIGRATION_FEATURE_IDS,
+    // each counter representing the number of times its corresponding feature ID is
+    // referenced in this transaction.
+    migrating_builtin: Vec<u16>,
```

Comment:
Can we do something like this? We **REALLY** do not want allocations per transaction.

Any mismatch is size is still caught by tests with the debug assert in `default()`

```rust
const NUM_MIGRATING_BUILTINS: usize = 3;

#[cfg_attr(test, derive(Eq, PartialEq))]
#[cfg_attr(feature = "dev-context-only-utils", derive(Clone))]
#[derive(Debug)]
struct MigrationBuiltinFeatureCounter {
    // The vector of counters, matching the size of the static vector MIGRATION_FEATURE_IDS,
    // each counter representing the number of times its corresponding feature ID is
    // referenced in this transaction.
    migrating_builtin: [u16; NUM_MIGRATING_BUILTINS],
}

impl Default for MigrationBuiltinFeatureCounter {
    fn default() -> Self {
        debug_assert_eq!(
            NUM_MIGRATING_BUILTINS,
            MIGRATION_FEATURES_ID.len(),
            "NUM_MIGRATING_BUILTINS must match the size of MIGRATION_FEATURES_ID"
        );
        Self {
            migrating_builtin: [0; NUM_MIGRATING_BUILTINS],
        }
    }
}
```

Reply:
553a77df12a4ea7d7e5f9d7e9ae6fade078b8322